R/repeat_linter.R

```diff
@@ -0,0 +1,46 @@
+#' Repeat linter
+#'
+#' Check that `while (TRUE)` is not used for infinite loops.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "while (TRUE) { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' # okay
+#' lint(
+#'   text = "repeat { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' @evalRd rd_tags("repeat_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+repeat_linter <- function() {
+  xpath <- "
+  //*[(self::WHILE or self::expr[SYMBOL_FUNCTION_CALL[.='`while`']])
```

Comment:
`//*` is really troublesome for XPath -- they are evaluated greedily and so this linter will be really expensive on files with a lot of expressions. See https://github.com/r-lib/lintr/wiki/Writing-robust,-performant-linters.

I really think we can ignore the `` `while` `` case unless there's a specific request for it -- that's a really unusual way to write a `while` loop. Otherwise, we would usually write this as `//WHILE[...] | //SYMBOL_FUNCTION_CALL[text() = '`while`']/parent::expr[...]`.

(TIL you can use `. = '...'` to check the text on a node btw! Nice find. I still like `text() = '...'` for expressiveness, `. = '...'` is a little bit harder to reason about)

Reply:
I will ignore the `` `while` `` case. 
I misunderstood you I thought you would have wished for this case by mentioning it as a possible test  [here](https://github.com/r-lib/lintr/pull/2107#discussion_r1316290069)